analysis/3_coregister-and-source-recon-bulk.py

```diff
@@ -0,0 +1,240 @@
+                                    subject = sub_run_combo,
+                                    filename = f"{recon_dir}/{sub_run_combo}/rhino/coreg/coreg_display_plot.html") # saves an interactive html plot at this location
+
+    # Compute forward model
+    gridstep = 10
+    source_recon.rhino.forward_model(
+        recon_dir,
+        sub_run_combo,
+        model="Single Layer",
+        gridstep=gridstep,
+    )
+
+    # view results
+    source_recon.rhino.bem_display(
+        recon_dir,
+        sub_run_combo,
+        display_outskin_with_nose=False,
+        display_sensors=True,
+        plot_type="surf",
+        filename=f"{recon_dir}/{sub_run_combo}/rhino/coreg/bem_display_plot.html",
+    )
+
+    from mne import read_forward_solution
+
+    # load forward solution
+    fwd_fname = op.join(recon_dir, sub_run_combo, "rhino", "model-fwd.fif") 
+    # tutorial said source_recon.rhino.get_coreg_filenames(recon_dir, subjects[0])["forward_model_file"]
+    # but this did not return a match for "forward_model_file"
+    print(fwd_fname)
+
+    fwd = read_forward_solution(fwd_fname)
+    leadfield = fwd["sol"]["data"]
+    print("Leadfield size : %d sensors x %d dipoles" % leadfield.shape)
+
+    import mne
+
+    # Temporal filtering
+
+    chantypes = ["grad", "mag"] 
+
+    # Get and setup the data
+    data = mne.io.read_raw_fif(fif_file, preload=True)
+    data = data.pick(chantypes)
+
+    # Filter to the beta band
+    print("Temporal Filtering")
+    data = data.filter(
+        l_freq=1,
+        h_freq=30,
+        method="iir",
+        iir_params={"order": 5, "btype": "bandpass", "ftype": "butter"},
+    )
+    print("Completed")
+
+    # Compute BEAMFORMER WEIGHTS
+    from osl.source_recon import rhino, beamforming, parcellation
+      
+    # Make LCMV beamformer filters
+    # Note that this will exclude any bad time segments when calculating the beamformer filters
+    filters = beamforming.make_lcmv(
+        recon_dir,
+        sub_run_combo,
+        data,
+        chantypes,
+        pick_ori="max-power-pre-weight-norm",
+        rank={"meg": 60},
+    )
+
+    print("Applying beamformer spatial filters")
+
+    # stc is source space time series (in head/polhemus space).
+    stc = beamforming.apply_lcmv(data, filters)
+
+    # Convert from head/polhemus space to standard brain grid in MNI space
+    recon_timeseries_mni, reference_brain_fname, recon_coords_mni, _ = \
+            beamforming.transform_recon_timeseries(recon_dir, 
+                                                    sub_run_combo, 
+                                                    recon_timeseries=stc.data, 
+                                                    reference_brain="mni")
+
+    print("Completed")
+    print("Dimensions of reconstructed timeseries in MNI space is (dipoles x all_tpts) = {}".format(recon_timeseries_mni.shape))
```

Comment:
Nitpick: why not use a f-string as you've done in other places?
```suggestion
    print(f"Dimensions of reconstructed timeseries in MNI space is (dipoles x all_tpts) = {recon_timeseries_mni.shape}")
```